Add restic_mode variable

jkirk opened the pull request:
It should be possible to allow all users to run the restic binary.
For that the variable restic_mode should be set to 0755 to set the
permissions of the file accordingly.

paulfantom commented:
What would be the purpose of that?

jkirk commented:
I sometimes want to allow all my users on the system to use the restic binary (mostly Linux Desktop systems). I then need `restic` to have executable right for `other`. (Like rsync for example).

jkirk commented:
(Sorry for the delay, somehow missed your answer).

I think I get your point, but in our scenario we (the admins) would like to be able to run `restic snapshots` and the like without sudo permissions.

Or do you think that allowing restic to be run as "ordinary" user is somehow risky? (Changing the mode of the binary is completely optional after all).

paulfantom commented:
I don't see any risk in running restic binary as an ordinary user and changing binary permissions to 0755 seems ok to me. @TheLastProject what do you think?

TheLastProject commented:
cap_dac_read_search+ep is set currently, so that would give ANY user read access to ANY file. So we definitely shouldn't unless we make sure that's not set

jkirk commented:
@TheLastProject Sorry, you lost me here. Why / Where are these capabilities set? Is this somehow specific to restic itself (shall we file a security bug there)?

TheLastProject commented:
It's set here: https://github.com/paulfantom/ansible-restic/blob/8fc1da0d33cd89f7ed879d7b4b2467532d765ac1/tasks/install.yml#L77-L82

Normally, it improves security, because restic could run as another user than root but still back up every file, see https://github.com/paulfantom/ansible-restic#security. However, if you give everyone execute permission, this becomes a serious security concern. I'm all for supporting a way to give everyone execute permission, but we need to make sure to unset it then.

jkirk commented:
Hey @paulfantom, @TheLastProject,

sorry for the very long delay.

What about only setting the capability when the "other" execute permission is not set? (I updated my PR (also rebased it on latest master)).

TheLastProject commented:
Well, you'd also want to unset it if that bit is set, otherwise doing a rollout and then deciding "oh nvm I actually want to allow everyone to use it" will still open you to the same security issue.

jkirk commented:
Oh, yes... Good point! I'll update my PR...

jkirk commented on tasks/install.yml:
```diff
@@ -80,6 +85,7 @@
   when:
     - ansible_os_family | lower == "debian"
     - restic_user != 'root'
+    - restic_group != 'root'
```
@TheLastProject I think this line is wrong too, right? `libcap2-bin` should be installed if restic_user OR restic_group are not 'root', like this:
```
 - restic_user != 'root' or restic_group != 'root'
```

TheLastProject replied:
I think this is @paulfantom's code, but it seems to me that that indeed should be changed.

jkirk commented on tasks/install.yml:
```diff
@@ -88,4 +94,6 @@
     state: present
   when:
     - restic_user != 'root'
+    - restic_group != 'root'
```
Same as above, right?

TheLastProject replied:
Same as above.

jkirk commented on tasks/install.yml:
```diff
@@ -88,4 +94,6 @@
   when:
     - restic_user != 'root'
+    - restic_group != 'root'
     - not ansible_check_mode
```
@TheLastProject This does not come from my PR, but still: what is this line for? The tasks are always run when _not ansible_check_mode_, no? Did I make a mistake in my thinking?

jkirk replied:
.. and skipped when in Check/Test-Mode.

TheLastProject replied:
I think it's just to not fail if the capabilities are not set

jkirk commented:
And: shouldn't we also unset the capability when owner and group are 'root' (again)?

(I so sorry for this comment flood... )

TheLastProject commented:
> And: shouldn't we also unset the capability when owner and group are 'root' (again)?
> 
> (I so sorry for this comment flood... )

Well, yup, we talked about that:

>Well, you'd also want to unset it if that bit is set, otherwise doing a rollout and then deciding "oh nvm I actually want to allow everyone to use it" will still open you to the same security issue.
